packages/admin-panel-server/src/routes/ImportDashboardVisualisationRoute.ts

```diff
@@ -4,18 +4,36 @@
  *
  */
 
+import assert from 'assert';
 import { Request, Response, NextFunction } from 'express';
 import fs from 'fs';
 
 import { Route } from '@tupaia/server-boilerplate';
-import { readJsonFile, UploadError } from '@tupaia/utils';
+import { readJsonFile, reduceToDictionary, snakeKeys, UploadError, yup } from '@tupaia/utils';
 
 import { MeditrakConnection } from '../connections';
 import {
+  dashboardSchema,
+  dashboardRelationObjectSchema,
   DashboardVisualisationExtractor,
   draftDashboardItemValidator,
   draftReportValidator,
 } from '../viz-builder';
+import {
+  Dashboard,
+  DashboardRecord,
+  DashboardRelationObject,
+  DashboardRelationRecord,
+} from '../viz-builder/types';
+
+const importFileSchema = yup.object().shape(
+  {
+    dashboards: yup.array().of(dashboardSchema),
+    dashboardRelations: yup.array().of(dashboardRelationObjectSchema),
```

Comment:
@rohan-bes I decided to simplify things here, and allow dashboards and relations to be specified on their own. I suspect that most users will just export/import using the admin panel, but "power" users will be able to do some sophisticated operations by updating the import file, eg

1. A user wants to update a viz and its dashboard, and migrate the updates across instances
2. The user has created some new temporary relations in instance A (don't ask me why 😄 ), and doesn't want to copy them across instance B
3. The solution: delete `dashboardRelations` from the import file

Another example:
1. The user exports a viz, including a new dashboard
2. After the fact, they realize they also want to also attach the viz to an existing dashboard
3. They can simply add a new relation entry in the config file

Users will be warned by error messages in case any required key is missing

Reply:
@kael89 I like this! Usually I just create one or two relations when testing in local, even if the dashboard item requires 10+ in deployment. So being able to add/remove relations from the file is very handy!